### PR TITLE
Add `wp` as a developer dependency

### PR DESCRIPTION
**Pull request by d-alleyne**
### Changes proposed in this Pull Request:

This Pull Request introduces an explicit declaration of the `wp` dependency, which improves the reliability of our translation file update process during releases.

In our previous setup, the successful operation hinged on the assumption that developers had `wp` pre-installed on their systems, as suggested by the [installation guide](https://wp-cli.org/#installing). This approach, however, introduced an unnecessary dependency on the individual developer's system environment.

In line with [best practices](https://12factor.net/dependencies), we are now clearly stating that `wp` is required. This new method eliminates potential issues tied to individual system setups and offers a smoother experience for all developers.

Relates to https://github.com/Automattic/i18n-issues/issues/594 .

- [x] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Execute `npm run prepare-release`, and attempt the first 3 steps
   ```
   1. Bump the version number in the composer.json file and the wc-calypso-bridge.php file.
   2. Update the changelog in readme.txt.
   3. Update the translation files (optional).
   ```
3. Attempt to download the translation files.
6. Verify that the translation files have been updated.
7. Once successful, abort, and delete the branch that was created.
8. Modify the `vendor/bin/wp i18n make-pot` [command](https://github.com/Automattic/wc-calypso-bridge/blob/master/scripts/commands/update-translations.js#L52) to point to a non-existent source directory. e.g Change line 53 of `scripts/commands/update-translations.js` to `vendor/bin/wp i18n make-pot ./non-existent-directory ${ POT_FILE_PATH } --ignore-domain`
9. Attempt steps 1-2.
10. Verify that it failed, and the output terminal describes the error correctly.
11. Delete the branch that was created.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: scripts/commands/update-translations.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ export default async function updateTranslations() {
 
 		// If the `wp i18n make-pot` command generates an error, an exception is thrown
 		const makePotResult = await execPromise(
-			`wp i18n make-pot . ${ POT_FILE_PATH } --ignore-domain`
+			`vendor/bin/wp i18n make-pot . ${ POT_FILE_PATH } --ignore-domain`
```
Comment (markbiek) [inline scripts/commands/update-translations.js:53]: Do we want to add a check that `vendor/bin/wp` exists?

Comment (d-alleyne) [reply scripts/commands/update-translations.js]: Done.

File: scripts/commands/update-translations.js [hunk 2]
```diff
@@ -44,13 +44,19 @@ async function verifyLangData( langData ) {
...
 	try {
+		if ( ! fs.existsSync( wpPath ) ) {
+			warning( 'wp not found, running composer install...' );
+			await execPromise( 'composer install' );
+		}
+
 		info( 'Executing wp i18n make-pot command...' );
 
 		// If the `wp i18n make-pot` command generates an error, an exception is thrown
 		const makePotResult = await execPromise(
```
Comment (markbiek) [inline scripts/commands/update-translations.js:58]: One more minor nit. It doesn't look like we're using `makePotResult` anywhere so we can probably delete that variable.

**Review by markbiek — COMMENTED** [review]
Overall, I think this is functioning as expected. I like that it runs `composer install` automatically if the `wp` binary isn't found.

Left one more very minor nit.

File: scripts/commands/update-translations.js (hunk 2 above)
Comment (d-alleyne) [reply scripts/commands/update-translations.js]: Done

**Review by markbiek — APPROVED** [review]
LGTM!

**Comment by d-alleyne** [comment]
Continued in https://github.com/Automattic/wc-calypso-bridge/pull/1254

**Pull request merged**